pkg/cmd/upgrade/upgrade_boot.go

```diff
@@ -312,14 +318,55 @@ func (o *UpgradeBootOptions) updateBootConfig(versionStreamURL string, versionSt
 		return errors.Wrapf(err, "failed to fetch upgrade tag %s from %s", upgradeVersion, bootConfigURL)
 	}
 
+	// Set up custom merge driver to ensure that specified files always use the local/dev env version in merges/cherry picks with conflicts
+	err = o.configureGitMergeExcludes()
+	if err != nil {
+		return errors.Wrapf(err, "configuring files (%s) to always use existing version when conflicting with upstream", strings.Join(filesExcludedFromCherryPick, ", "))
+	}
+
 	err = o.cherryPickCommits(configCloneDir, currentSha, upgradeSha)
 	if err != nil {
 		return errors.Wrap(err, "failed to cherry pick upgrade commits")
 	}
-	err = o.excludeFiles(currentSha)
+	return nil
+}
+
+func (o *UpgradeBootOptions) configureGitMergeExcludes() error {
+	err := o.Git().Config(o.Dir, "--local", fmt.Sprintf("merge.%s.name", keepDevEnvKey), keepDevEnvDriverName)
+	if err != nil {
+		return errors.Wrapf(err, "configuring name for exclude merge driver in %s", o.Dir)
+	}
+	err = o.Git().Config(o.Dir, "--local", fmt.Sprintf("merge.%s.driver", keepDevEnvKey), "true")
+	if err != nil {
+		return errors.Wrapf(err, "configuring driver for exclude merge driver in %s", o.Dir)
+	}
+
+	// Read the existing .git/info/attributes if it exists
```

Comment:
I'd say no for two reasons: this is the only place we're doing it currently, and `gitter` feels to me like it should only be for `git` commands, and this isn't that.

Reply:
> I'd say no for two reasons: this is the only place we're doing it currently,

right, but say someone else needs to do the same, where would he naturally be looking for it?

> gitter feels to me like it should only be for git commands, and this isn't that.

It is already more than pure git command execution. There are functions in there already dealing with .git internas. 
But since you feel so strongly about it, I resy my case.

Reply:
I also can't think of a good abstraction, but lemme try to bang something out.

Reply:
btw, another argument for moving it would be to write some tests for it in the gitter area.